src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -159,6 +160,73 @@ public ComplaintResponse createComplaintResponseRepresentingLock(Complaint compl
         return persistedComplaintResponse;
     }
 
+    /**
+     * Resolves a complaint by filling in the empty complaint response attached to it
+     *
+     * The empty complaint response acts as a lock. Only the creator of the empty complaint response and instructors can resolve empty complaint response as long as the lock
+     * is running. For lock duration calculation see: {@link ComplaintResponse#isCurrentlyLocked()}. These methods fill in the initial complaint response and either accepts
+     * or denies the associated complaint, thus resolving the complaint
+     *
+     * @param updatedComplaintResponse complaint response containing the information necessary for resolving the complaint
+     * @param complaintResponseId      complaint response id
+     * @return complaintResponse of resolved complaint
+     */
+    public ComplaintResponse resolveComplaint(ComplaintResponseUpdateDTO updatedComplaintResponse, Long complaintResponseId) {
+        if (complaintResponseId == null) {
+            throw new IllegalArgumentException("The complaint response needs to have an id");
+        }
+        Optional<ComplaintResponse> complaintResponseFromDatabaseOptional = complaintResponseRepository.findById(complaintResponseId);
+        if (complaintResponseFromDatabaseOptional.isEmpty()) {
+            throw new IllegalArgumentException("The complaint response was not found in the database");
+        }
+        ComplaintResponse emptyComplaintResponseFromDatabase = complaintResponseFromDatabaseOptional.get();
+        if (emptyComplaintResponseFromDatabase.getSubmittedTime() != null || emptyComplaintResponseFromDatabase.getResponseText() != null) {
+            throw new IllegalArgumentException("The complaint response is not empty");
+        }
+        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(emptyComplaintResponseFromDatabase.getComplaint().getId());
+        if (originalComplaintOptional.isEmpty()) {
+            throw new IllegalArgumentException("The complaint was not found in the database");
+        }
+        Complaint originalComplaint = originalComplaintOptional.get();
+        if (originalComplaint.isAccepted() != null) {
```

Comment:
Consider moving all gates that make sure that the complaint isn't already resolved to a distinct method and only check here whether the response is valid.